packages/backend/src/auth/auth.service.ts

```diff
@@ -72,58 +73,74 @@ export const signup = async (req: express.Request, res: express.Response) => {
   }
 }
 
-export const login = async (req: express.Request, res: express.Response) => {
-  const { email, password } = await zParse(loginSchema, req)
+export const login = async (
+  req: express.Request,
+  res: express.Response,
+  next: NextFunction
+) => {
+  try {
+    const { email, password } = await zParse(loginSchema, req)
 
-  const user = await User.query()
-    .findOne({ email })
-    .select('id', 'email', 'password')
-    .throwIfNotFound(new UnauthorisedException('Invalid credentials'))
```

Comment:
This did not work. It throws a NotFound Error, not an Unauthorised Exception. The object passed here to the throwIfNotFound method only adds it to the "data" property of the NotFoundError. It does not replace the instance with an UnauthroisedException one.

The only way to change this behaviour is by doing it globally on the knex Model instance in index.ts. We can define a custom class there to be thrown instead of NotFound. But I'd rather just not use the feature at all, and just check if the user exists or not as I did in this change request